src/main/java/org/zalando/nakadi/service/StorageService.java

```diff
@@ -28,16 +35,40 @@
 @Service
 public class StorageService {
 
+    public static final String ZK_TIMELINES_DEFAULT_STORAGE = "/nakadi/timelines/default_storage";
     private static final Logger LOG = LoggerFactory.getLogger(StorageService.class);
-
     private final ObjectMapper objectMapper;
     private final StorageDbRepository storageDbRepository;
+    private final DefaultStorage defaultStorage;
+    private final CuratorFramework curator;
 
     @Autowired
     public StorageService(final ObjectMapper objectMapper,
-                          final StorageDbRepository storageDbRepository) {
+                          final StorageDbRepository storageDbRepository,
+                          @Qualifier("default_storage") final DefaultStorage defaultStorage,
+                          final ZooKeeperHolder zooKeeperHolder) {
         this.objectMapper = objectMapper;
         this.storageDbRepository = storageDbRepository;
+        this.defaultStorage = defaultStorage;
+        this.curator = zooKeeperHolder.get();
+    }
+
+    @PostConstruct
+    private void watchDefaultStorage() {
+        try {
+            curator.getData().usingWatcher((CuratorWatcher) event -> {
+                final byte[] defaultStorageId = curator.getData().forPath(ZK_TIMELINES_DEFAULT_STORAGE);
+                if (defaultStorageId != null) {
+                    final Result<Storage> storageResult = getStorage(new String(defaultStorageId));
+                    if (storageResult.isSuccessful()) {
+                        defaultStorage.setStorage(storageResult.getValue());
+                    }
+                }
+                watchDefaultStorage();
+            }).forPath(ZK_TIMELINES_DEFAULT_STORAGE);
+        } catch (final Exception e) {
+            LOG.warn(e.getMessage(), e);
```

Comment:
maybe you can add some context here in the log. Like "Error occurred when updating default storage info" or something like that.

Reply:
Actually, you can see that by just checking the stacktrace in the logs